### PR TITLE
Contrib help

### PR DESCRIPTION
**Pull request by Hainish**
cc @jeremyn 
Still a work in progress

### CODE REVIEW

**Comment by jeremyn** [comment]
Glad this is getting done. What feedback do you want from me?

**Comment by Hainish** [comment]
@jeremyn if you want to link to all the related issues that'd be helpful.  Also: do you think merging the ruleset style guide into `CONTRIBUTING.md` is a good idea?  I was thinking it's similar subject matter and might make sense, but the style guide is also pretty long and I fear cluttering `CONTRIBUTING.md`, which ideally would be concise and easily navigable.

**Comment by jeremyn** [comment]
I've been using https://github.com/EFForg/https-everywhere/issues/7707 as a collection issue for documentation stuff, including my lengthy comment at https://github.com/EFForg/https-everywhere/issues/7707#issuecomment-263462800. So please consider linking that here as the same as linking to all the different discussions.

I think everything should be merged into `CONTRIBUTING.md`, including the style guide and the FAQ, as I mention at the top of that comment.

**Comment by Hainish** [comment]
I also have https://github.com/efforg/https-everywhere/issues/7662 as a relevant ref.

**Comment by jeremyn** [comment]
#7662 is linked from #7707.

**Comment by Hainish** [comment]
I think this is finally in a good place to look at and review.

I've done my best to combine the existing `ruleset-styleguide.md` with updated considerations and comments in #7707, as well as grouping and categorizing topics logically for ease of reading.

@jeremyn care to take a look?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,11 +1,310 @@
...
+
+# Contributing Code
+
+We encourage code contributions to HTTPS Everywhere.  There are a few considerations to keep in mind when contributing code.
+
+Officially supported browsers:
+
+- Firefox Stable
+- Firefox ESR
+- Firefox Dev
```
Comment (J0WI) [inline CONTRIBUTING.md]: To what release channel is "Firefox Dev" referring to? There are Beta, Aurora (aka. Dev Edition) and Nightly.

**Comment by J0WI** [comment]
I like the new documentation, thanks for merging them!
What I concern is that we now have a huge `CONTRIBUTING.md` compared to other projects. So the question is how carefully this will be read by new contributors.

File: CONTRIBUTING.md (hunk 1 above)
Comment (Hainish) [inline CONTRIBUTING.md]: I guess we should remove the dev channel from this documentation altogether.  We do test with Firefox Aurora, but this is to detect potential issues coming down the line.  It isn't actually officially supported.

**Comment by Hainish** [comment]
@J0WI a valid concern.  I think it's useful to have somewhere for ruleset maintainers to point to when asking contribs to clean up a PR, but I don't want to indicate that every contrib should read the full guide before getting their hands dirty.  Maybe we can say as much as an intro to the guide.

**Comment by Hainish** [comment]
@jgillula will also help by lending a set of fresh eyes to this documentation.

**Comment by Hainish** [comment]
I think this is in a good state now, and @jgillula has reviewed.  We can build this out further but that shouldn't prevent us from pulling this in this now, so I'm going to go ahead and merge.

**Comment by jgillula** [comment]
Wait, the document I sent wasn't complete. In particular, there's still a FIXME in there...

**Comment by Hainish** [comment]
@jeremyn I'll resolve that now

**Comment by jeremyn** [comment]
Thanks for doing this @Hainish!

**Pull request merged**